gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTable.java

```diff
@@ -137,6 +137,10 @@ else if (colIndex instanceof String) {
     double[] colArray = new double[col.size()];
     Conversions conversionMethod = null;
     for (int i = 0; i < col.size(); i++) {
+      Object v = col.get(i);
+      if (v instanceof String && ((String) v).trim().isEmpty()) {
+        col.set(i, "0");
```

Comment:
Why `0` is the default in case the column has no value? Reading the PR description I know you are expecting numbers in these columns, but it could be any kind of data in other scenarios `DATE`, `TEXT`,...
What if - to make it more similar to how relational DBs look like - you use `NULL`?

Reply:
It is explicitly expecting numeric data based on the use of this method. Null and empty string need to be considered 0 in these cases for the conversion to make sense.